pkg/operation/botanist/component/dependencywatchdog/dependency_watchdog.go

```diff
@@ -362,6 +362,17 @@ func (d *dependencyWatchdog) clusterRoleRules() []rbacv1.PolicyRule {
 				Resources: []string{"deployments", "deployments/scale"},
 				Verbs:     []string{"get", "list", "watch", "update"},
 			},
+			{
+				APIGroups: []string{"coordination.k8s.io"},
+				Resources: []string{"leases"},
+				Verbs:     []string{"create"},
+			},
+			{
+				APIGroups:     []string{"coordination.k8s.io"},
+				ResourceNames: []string{"dependency-watchdog-probe"},
```

Comment:
What about `dependency-watchdog-endpoint`, why only `dependency-watchdog-probe`'s RBAC is adapted.
For reference `dependency-watchdog-endpoing@v0.7.0` fails to start with:
```
E0301 13:03:57.498985       1 leaderelection.go:331] error retrieving resource lock garden/dependency-watchdog: leases.coordination.k8s.io "dependency-watchdog" is forbidden: User "system:serviceaccount:garden:dependency-watchdog-endpoint" cannot get resource "leases" in API group "coordination.k8s.io" in the namespace "garden"
E0301 13:04:00.181425       1 leaderelection.go:331] error retrieving resource lock garden/dependency-watchdog: leases.coordination.k8s.io "dependency-watchdog" is forbidden: User "system:serviceaccount:garden:dependency-watchdog-endpoint" cannot get resource "leases" in API group "coordination.k8s.io" in the namespace "garden"
```

Reply:
That was a miss on my end, I didn't check the endpoint watchdog and was too focussed on probe handling. 
Fixed with the [commit](https://github.com/gardener/gardener/pull/5497/commits/5d13f962b5697dfe2897b3e605905c66410b27f6)